colcon_core/verb/__init__.py

```diff
@@ -124,6 +124,41 @@ def check_and_mark_install_layout(install_base, *, merge_install):
     marker_path.write_text(this_install_layout + '\n')
 
 
+def check_and_mark_root_dir(this_build_tool='colcon'):
+    """
+    Check the marker file for root workspace.
+
+    If marker file is found in parent directory, raise error;
+    if it is found in current direcotry, continue to build.
+    If no marker file found, create marker file.
+
+    The marker filename is `.colcon`.
+
+    :param str this_build_tool: The name of this build tool
+    :raises RuntimeError: if marker file is found in parent directory
+    """
+    original_path = Path(os.getcwd())
```

Comment:
Wouldn't it be clearer (easier to test, and more reusable) if the function would have the start path as an argument rather then getting it internally from global state?